src/shared/elf-util.c

```diff
@@ -532,14 +532,20 @@ static int module_callback(Dwfl_Module *mod, void **userdata, const char *name,
         for (size_t i = 0; i < n_program_headers; ++i) {
                 GElf_Phdr mem, *program_header;
                 Elf_Data *data;
+                GElf_Addr end_of_segment;
 
                 /* The core file stores the ELF files in the PT_LOAD segment. */
                 program_header = sym_gelf_getphdr(elf, i, &mem);
                 if (!program_header || program_header->p_type != PT_LOAD)
                         continue;
 
+                /* Check that the end of segment is a valid address. */
+                if (__builtin_add_overflow(program_header->p_vaddr, program_header->p_memsz, &end_of_segment))
+                        /* The core seems corrupted. */
+                        return DWARF_CB_ABORT;
```

Comment:
The patch is OK, but please add an error message. In the other places where `return DWARF_CB_ABORT` is used, the helper function already logged before, but we'd exit silently.

Reply:
Ok, I have just pushed a modified version.